components/backup-stream/src/utils.rs

```diff
@@ -627,6 +632,81 @@ impl AsyncRead for FilesReader {
     }
 }
 
+#[async_trait::async_trait]
+pub trait CompressionWriter: Sync + Send + 'static {
```

Comment:
Can we make it a sub trait of `AsyncWrite`?

Reply:
Making it a subtrait of `AsyncWrite` can eliminate a boxed future, and I think `write_all` would be a hot function.